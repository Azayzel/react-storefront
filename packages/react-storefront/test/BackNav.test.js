/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import TestProvider from './TestProvider'
import { createMemoryHistory } from 'history'
import BackNav from '../src/BackNav'
import AppModelBase from '../src/model/AppModelBase'
import SearchResultsModelBase, {
  LAYOUT_LIST,
  LAYOUT_GRID
} from '../src/model/SearchResultsModelBase'

describe('BackNav', () => {
  let app, history

  beforeEach(() => {
    app = AppModelBase.create({})
    history = createMemoryHistory({
      initialEntries: ['/c/2']
    })
  })

  it('renders', () => {
    expect(
      mount(
        <TestProvider history={history} app={app}>
          <BackNav text="Rugs" url="/s/1" />
        </TestProvider>
      )
    ).toMatchSnapshot()
  })

  it('renders the view type switcher', () => {
    const searchResults = SearchResultsModelBase.create({ id: '1' })

    expect(
      mount(
        <TestProvider history={history}>
          <BackNav text="Furniture" url="/c/1" searchResults={searchResults} />
        </TestProvider>
      )
    ).toMatchSnapshot()
  })

  it('render the corresponding tag if `labelComponent` prop provided', () => {
    expect(
      mount(
        <TestProvider history={history}>
          <BackNav text="Rugs" url="/rugs/1" labelComponent="h1" />
        </TestProvider>
      )
    ).toMatchSnapshot()
  })

  it('goes to the specified url when clicked', () => {
    const wrapper = mount(
      <TestProvider history={history}>
        <BackNav text="Rugs" url="/c/1" />
      </TestProvider>
    )

    wrapper
      .find('span[onClick]')
      .at(0)
      .simulate('click')
    expect(history.location.pathname).toBe('/c/1')
  })

  it('goes back in history when no url is provided', () => {
    const wrapper = mount(
      <TestProvider history={history}>
        <BackNav text="Back" />
      </TestProvider>
    )

    wrapper
      .find('span[onClick]')
      .at(0)
      .simulate('click')
    expect(history.location.pathname).toBe('/c/2')
  })

  it('switches the view types to grid when clicked', () => {
    const searchResults = SearchResultsModelBase.create({ id: '1', layout: LAYOUT_LIST })

    const wrapper = mount(
      <TestProvider history={history}>
        <BackNav text="Back" searchResults={searchResults} />
      </TestProvider>
    )

    wrapper
      .find('BorderAllIcon[onClick]')
      .at(0)
      .simulate('click')
    expect(searchResults.layout).toBe(LAYOUT_GRID)
  })

  it('switches the view types to list when clicked', () => {
    const searchResults = SearchResultsModelBase.create({ id: '1', layout: LAYOUT_GRID })

    const wrapper = mount(
      <TestProvider history={history}>
        <BackNav text="Back" searchResults={searchResults} />
      </TestProvider>
    )

    wrapper
      .find('ViewAgendaIcon[onClick] svg')
      .at(0)
      .simulate('click')
    expect(searchResults.layout).toBe(LAYOUT_LIST)
  })
})
