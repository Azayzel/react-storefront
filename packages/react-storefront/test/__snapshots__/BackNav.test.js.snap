// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackNav render the corresponding tag if \`labelComponent\` prop provided 1`] = `
<TestProvider history={{...}}>
  <Provider router={{...}} app={{...}} history={{...}} nextId={[Function: nextId]}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(inject-BackNav-with-router) text="Rugs" url="/rugs/1" labelComponent="h1">
        <inject-BackNav-with-router text="Rugs" url="/rugs/1" labelComponent="h1" classes={{...}}>
          <BackNav text="Rugs" url="/rugs/1" labelComponent="h1" classes={{...}} router={{...}}>
            <WithStyles(Paper) className="RSFBackNav-root-155">
              <Paper className="RSFBackNav-root-155" classes={{...}} component="div" elevation={2} square={false}>
                <div className="MuiPaper-root-161 MuiPaper-elevation2-165 MuiPaper-rounded-162 RSFBackNav-root-155">
                  <WithStyles(Typography) variant="caption">
                    <Typography variant="caption" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                      <span className="MuiTypography-root-188 MuiTypography-caption-198">
                        <span onClick={[Function: onClick]} className="RSFBackNav-backButtonWrapper-157">
                          <pure(KeyboardArrowLeftIcon)>
                            <KeyboardArrowLeftIcon>
                              <WithStyles(SvgIcon)>
                                <SvgIcon classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                  <svg className="MuiSvgIcon-root-223" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                    <path d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z" />
                                    <path fill="none" d="M0 0h24v24H0V0z" />
                                  </svg>
                                </SvgIcon>
                              </WithStyles(SvgIcon)>
                            </KeyboardArrowLeftIcon>
                          </pure(KeyboardArrowLeftIcon)>
                        </span>
                        <h1 className="RSFBackNav-label-156">
                          Rugs
                        </h1>
                      </span>
                    </Typography>
                  </WithStyles(Typography)>
                </div>
              </Paper>
            </WithStyles(Paper)>
          </BackNav>
        </inject-BackNav-with-router>
      </WithStyles(inject-BackNav-with-router)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;

exports[`BackNav renders 1`] = `
<TestProvider history={{...}} app={{...}}>
  <Provider router={{...}} app={{...}} history={{...}} nextId={[Function: nextId]}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(inject-BackNav-with-router) text="Rugs" url="/s/1">
        <inject-BackNav-with-router text="Rugs" url="/s/1" classes={{...}}>
          <BackNav text="Rugs" url="/s/1" classes={{...}} router={{...}} labelComponent="span">
            <WithStyles(Paper) className="RSFBackNav-root-1">
              <Paper className="RSFBackNav-root-1" classes={{...}} component="div" elevation={2} square={false}>
                <div className="MuiPaper-root-7 MuiPaper-elevation2-11 MuiPaper-rounded-8 RSFBackNav-root-1">
                  <WithStyles(Typography) variant="caption">
                    <Typography variant="caption" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                      <span className="MuiTypography-root-34 MuiTypography-caption-44">
                        <span onClick={[Function: onClick]} className="RSFBackNav-backButtonWrapper-3">
                          <pure(KeyboardArrowLeftIcon)>
                            <KeyboardArrowLeftIcon>
                              <WithStyles(SvgIcon)>
                                <SvgIcon classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                  <svg className="MuiSvgIcon-root-69" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                    <path d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z" />
                                    <path fill="none" d="M0 0h24v24H0V0z" />
                                  </svg>
                                </SvgIcon>
                              </WithStyles(SvgIcon)>
                            </KeyboardArrowLeftIcon>
                          </pure(KeyboardArrowLeftIcon)>
                        </span>
                        <span className="RSFBackNav-label-2">
                          Rugs
                        </span>
                      </span>
                    </Typography>
                  </WithStyles(Typography)>
                </div>
              </Paper>
            </WithStyles(Paper)>
          </BackNav>
        </inject-BackNav-with-router>
      </WithStyles(inject-BackNav-with-router)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;

exports[`BackNav renders the view type switcher 1`] = `
<TestProvider history={{...}}>
  <Provider router={{...}} app={{...}} history={{...}} nextId={[Function: nextId]}>
    <MuiThemeProviderOld theme={{...}}>
      <WithStyles(inject-BackNav-with-router) text="Furniture" url="/c/1" searchResults={{...}}>
        <inject-BackNav-with-router text="Furniture" url="/c/1" searchResults={{...}} classes={{...}}>
          <BackNav text="Furniture" url="/c/1" searchResults={{...}} classes={{...}} router={{...}} labelComponent="span">
            <WithStyles(Paper) className="RSFBackNav-root-78">
              <Paper className="RSFBackNav-root-78" classes={{...}} component="div" elevation={2} square={false}>
                <div className="MuiPaper-root-84 MuiPaper-elevation2-88 MuiPaper-rounded-85 RSFBackNav-root-78">
                  <WithStyles(Typography) variant="caption">
                    <Typography variant="caption" theme={{...}} classes={{...}} align="inherit" color="default" gutterBottom={false} headlineMapping={{...}} noWrap={false} paragraph={false}>
                      <span className="MuiTypography-root-111 MuiTypography-caption-121">
                        <span onClick={[Function: onClick]} className="RSFBackNav-backButtonWrapper-80">
                          <pure(KeyboardArrowLeftIcon)>
                            <KeyboardArrowLeftIcon>
                              <WithStyles(SvgIcon)>
                                <SvgIcon classes={{...}} color="inherit" component="svg" fontSize="default" viewBox="0 0 24 24">
                                  <svg className="MuiSvgIcon-root-146" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation">
                                    <path d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z" />
                                    <path fill="none" d="M0 0h24v24H0V0z" />
                                  </svg>
                                </SvgIcon>
                              </WithStyles(SvgIcon)>
                            </KeyboardArrowLeftIcon>
                          </pure(KeyboardArrowLeftIcon)>
                        </span>
                        <span className="RSFBackNav-label-79">
                          Furniture
                        </span>
                        <span className="RSFBackNav-switchButtonsWrapper-81">
                          <span className="RSFBackNav-switchButton-82 RSFBackNav-selectedSwitchButton-83">
                            <pure(BorderAllIcon) color="secondary" onClick={[Function: onClick]}>
                              <BorderAllIcon color="secondary" onClick={[Function: onClick]}>
                                <WithStyles(SvgIcon) color="secondary" onClick={[Function: onClick]}>
                                  <SvgIcon color="secondary" onClick={[Function: onClick]} classes={{...}} component="svg" fontSize="default" viewBox="0 0 24 24">
                                    <svg className="MuiSvgIcon-root-146 MuiSvgIcon-colorSecondary-148" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation" onClick={[Function: onClick]}>
                                      <path d="M3 3v18h18V3H3zm8 16H5v-6h6v6zm0-8H5V5h6v6zm8 8h-6v-6h6v6zm0-8h-6V5h6v6z" />
                                      <path fill="none" d="M0 0h24v24H0z" />
                                    </svg>
                                  </SvgIcon>
                                </WithStyles(SvgIcon)>
                              </BorderAllIcon>
                            </pure(BorderAllIcon)>
                          </span>
                          <span className="RSFBackNav-switchButton-82">
                            <pure(ViewAgendaIcon) color="primary" onClick={[Function: onClick]}>
                              <ViewAgendaIcon color="primary" onClick={[Function: onClick]}>
                                <WithStyles(SvgIcon) color="primary" onClick={[Function: onClick]}>
                                  <SvgIcon color="primary" onClick={[Function: onClick]} classes={{...}} component="svg" fontSize="default" viewBox="0 0 24 24">
                                    <svg className="MuiSvgIcon-root-146 MuiSvgIcon-colorPrimary-147" focusable="false" viewBox="0 0 24 24" color={[undefined]} aria-hidden="true" role="presentation" onClick={[Function: onClick]}>
                                      <path fill="none" d="M0 0h24v24H0z" />
                                      <path d="M20 13H3c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h17c.55 0 1-.45 1-1v-6c0-.55-.45-1-1-1zm0-10H3c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1h17c.55 0 1-.45 1-1V4c0-.55-.45-1-1-1z" />
                                    </svg>
                                  </SvgIcon>
                                </WithStyles(SvgIcon)>
                              </ViewAgendaIcon>
                            </pure(ViewAgendaIcon)>
                          </span>
                        </span>
                      </span>
                    </Typography>
                  </WithStyles(Typography)>
                </div>
              </Paper>
            </WithStyles(Paper)>
          </BackNav>
        </inject-BackNav-with-router>
      </WithStyles(inject-BackNav-with-router)>
    </MuiThemeProviderOld>
  </Provider>
</TestProvider>
`;
